### ISSUE TITLE
Chroma db Code changed thats why unable to access the vectorstore from ChromaDB for embeddings

### ISSUE THREAD

**Issue by utkarshg1** [langchain-ai/langchain#19848]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

~~~python
# Python code
from langchain_openai import OpenAIEmbeddings
from langchain_community.vectorstores import Chroma
vectorstore = Chroma.from_documents(documents=splits,
                                    embedding=OpenAIEmbeddings(model="text-embedding-3-large"), 
                                    persist_directory = "./chroma_db")
~~~

### Error Message and Stack Trace (if applicable)

File /workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/chromadb/api/types.py:211, in validate_embedding_function(embedding_function)
    [208](https://vscode-remote+utkarshg1-002dopenailangcha-002dh4687x6hb59-002ews-002dus110-002egitpod-002eio.vscode-resource.gitpod.io/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/chromadb/api/types.py:208) protocol_signature = signature(EmbeddingFunction.__call__).parameters.keys()
    [210](https://vscode-remote+utkarshg1-002dopenailangcha-002dh4687x6hb59-002ews-002dus110-002egitpod-002eio.vscode-resource.gitpod.io/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/chromadb/api/types.py:210) if not function_signature == protocol_signature:
--> [211](https://vscode-remote+utkarshg1-002dopenailangcha-002dh4687x6hb59-002ews-002dus110-002egitpod-002eio.vscode-resource.gitpod.io/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/chromadb/api/types.py:211)     raise ValueError(
    [212](https://vscode-remote+utkarshg1-002dopenailangcha-002dh4687x6hb59-002ews-002dus110-002egitpod-002eio.vscode-resource.gitpod.io/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/chromadb/api/types.py:212)         f"Expected EmbeddingFunction.__call__ to have the following signature: {protocol_signature}, got {function_signature}\n"
    [213](https://vscode-remote+utkarshg1-002dopenailangcha-002dh4687x6hb59-002ews-002dus110-002egitpod-002eio.vscode-resource.gitpod.io/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/chromadb/api/types.py:213)         "Please see https://docs.trychroma.com/embeddings for details of the EmbeddingFunction interface.\n"
    [214](https://vscode-remote+utkarshg1-002dopenailangcha-002dh4687x6hb59-002ews-002dus110-002egitpod-002eio.vscode-resource.gitpod.io/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/chromadb/api/types.py:214)         "Please note the recent change to the EmbeddingFunction interface: https://docs.trychroma.com/migration#migration-to-0416---november-7-2023 \n"
    [215](https://vscode-remote+utkarshg1-002dopenailangcha-002dh4687x6hb59-002ews-002dus110-002egitpod-002eio.vscode-resource.gitpod.io/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/chromadb/api/types.py:215)     )

ValueError: Expected EmbeddingFunction.__call__ to have the following signature: odict_keys(['self', 'input']), got odict_keys(['args', 'kwargs'])
Please see https://docs.trychroma.com/embeddings for details of the EmbeddingFunction interface.
Please note the recent change to the EmbeddingFunction interface: https://docs.trychroma.com/migration#migration-to-0416---november-7-2023

### Description

I'm Trying to get Embeddings but looks like ChromaDB has changed functionality from their end.
Hence Langchain code also needs to be changed

https://docs.trychroma.com/embeddings/openai

### System Info

langchain==0.1.14
langchain-community==0.0.30
langchain-core==0.1.37
langchain-openai==0.1.1
langchain-text-splitters==0.0.1
langchainhub==0.1.15
chromadb==0.4.24

**Comment by wanjianwei** [comment]
i also met this error, and is looking for a solution

**Comment by srnsrn120** [comment]
> Same issue i am also facing .

If Any Work Around  is there please help

**Comment by liugddx** [comment]
Let me see

**Comment by avoutsas67** [comment]
Same issue with me - it seems the version 0.4.16 of ChromaDB introduced a breaking change.

**Comment by liugddx** [comment]
I tried the code and it works for me, what's the problem?

**Comment by avoutsas67** [comment]
@liugddx see the top message in the thread by @utkarshg1. Describes the issue quite clearly. 
Many thanks for looking into this.

**Comment by liugddx** [comment]
If you want a quick fix I have a solution


```
openai_ef = embedding_functions.OpenAIEmbeddingFunction(model_name="text-embedding-3-large",api_key=os.environ["OPENAI_API_KEY"])
vectorstore = Chroma.from_documents(documents=all_splits,
                                    embedding=openai_ef,
                                    persist_directory = "./chroma_db")
```

And change line [279](https://github.com/langchain-ai/langchain/blob/8711a05a5164195950666a7fe8ad0b62a822a320/libs/community/langchain_community/vectorstores/chroma.py#L279) of chroma.py to the following method

```
            embeddings = self._embedding_function(texts)

```

**Comment by utkarshg1** [comment]
> > Same issue i am also facing .
> 
> If Any Work Around is there please help

An easiest workaround for this is using the FAISS cpu as Vectorstore
Make sure you install faiss-cpu package as well
~~~cmd
pip install faiss-cpu
~~~

Python code below: 
~~~python
# Index
from langchain_openai import OpenAIEmbeddings
from langchain_community.vectorstores import FAISS
vectorstore = FAISS.from_documents(documents=splits,
                                   embedding=OpenAIEmbeddings())

# Retriever
vectorstore.as_retriever()
~~~

**Comment by srnsrn120** [comment]
> _embedding_function

No , Now i am using Multi vector retriever, i am not able to use Faiss or QdrantClient, only Chroma i am able to use 
So thats why i  wanted Chroma work around .

**Comment by baskaryan** [comment]
Should be fixed in `langchain-community==0.0.31`, please let me know if you're still seeing an issue
